app/src/main/java/ch/epfl/sdp/login/LoginFormActivity.java

```diff
@@ -26,10 +24,10 @@ protected void onCreate(Bundle savedInstanceState) {
         lemail = findViewById(R.id.emaillog);
         lpassword = findViewById(R.id.passwordlog);
 
-        authenticationAPI = DependencyFactory.getAuthenticationAPI();
+        authenticationAPI = ((MyApplication) getApplication()).appContainer.authenticationAPI;
 
         // If the user has already logged in, go to MainActivity directly
-        if(authenticationAPI.getCurrentUserEmail() != null) {
+        if (authenticationAPI.getCurrentUserEmail() != null) {
```

Comment:
Sorry, but I don't like null checking.

Reply:
Actually, that is not from me. I only did cntl+alt+L to reformat and the space came out. This comes from the branch where we wanted the user to stay online if he quit the app.

Reply:
Actually, in this case, it's normal the currentUserEmail can be null, as the user is in general not logged in when we are on the LoginFormActivity.